### ISSUE TITLE
Buttons stay disabled, when selection is released outside of editor

### ISSUE THREAD

**Issue by choffmeister** [guardian/scribe-plugin-toolbar#20]
# How to reproduce

1. Go to the current live example of scribe [here](http://guardian.github.io/scribe/).
2. Start selecting some text
3. While still pressing the mouse, move the mouse outside of the editor and release.

Now the buttons stay disabled.

![scribe-selection-bug](https://cloud.githubusercontent.com/assets/671620/10608641/3bcb06a4-7740-11e5-8a05-49b2a86c7c3d.gif)

# How to fix

The easy fix is this:

```diff
diff --git a/src/scribe-plugin-toolbar.js b/src/scribe-plugin-toolbar.js
index 36f0248..b42b66a 100644
--- a/src/scribe-plugin-toolbar.js
+++ b/src/scribe-plugin-toolbar.js
@@ -35,6 +35,11 @@ define(function () {
         scribe.el.addEventListener('focus', updateUi);
         scribe.el.addEventListener('blur', updateUi);

+        // Start selecting in the element but releasing the mouse
+        // outside of the element does not trigger the elements mouseup.
+        // So the buttons may remain disabled.
+        window.addEventListener('mouseup', updateUi);
+
         // We also want to update the UI whenever the content changes. This
         // could be when one of the toolbar buttons is actioned.
         scribe.on('content-changed', updateUi);
```

Wanted to create a PR for this, but did not manage to get the test suite running on my machine (using node 4.2.1).

**Comment by choffmeister** [comment]
Updated the original description, since as I noticed this only occurs in chrome. In firefox it works fine. Seems, like not the mouse releasing is the problem, but what happens while starting the selection. In firefox when starting the selection, all buttons are activated. In chrome not. Guess the problem is the `updateUi` while the first click.

**Comment by choffmeister** [comment]
When clicking the first time into the editor, the `var selection = new scribe.api.Selection();` returns the following:

* Chrome: `Selection {selection: Selection}`
* Firefox: `Object { selection: Selection, range: Range }`

Maybe the missing `range` is the problem?

**Comment by mindplay-dk** [comment]
I solved this - but at this point, my toolbar plugin is basically a complete rewrite.

I'm going to just dump it here, in case you're interested:

https://gist.github.com/mindplay-dk/f6d76ae9b7ff45983539

Several other things are different in my version of this plugin, because I have several shared toolbars (one fixed, and one floating) and lots of Scribe instances.

I use a single set of button event-handlers - attaching dedicated event-handlers for every instance of Scribe seemed unnecessary. 

I also avoid using an option and if-statement as a means of creating shared toolbars - instead, creating one or more instances is the consumer's choice, so you can choose to create a single instance of the toolbar plugin, with one set of button-handlers, etc. and apply the same plugin instance to several editors, e.g.:

```
var toolbar = scribePluginToolbar($('#toolbar').get(0));
scribe1.use(toolbar);
scribe2.use(toolbar);
// ...
```

In other words, you create a toolbar plugin for every toolbar - rather than a toolbar plugin for every editor instance, which may not be the "idiomatic" Scribe approach. But this approach should work well with any number of shared or dedicated toolbars.

Personally, I find this approach also leads to simpler and easier to understand code, but YMMV.

Note that I've paid no attention to backwards compatibility, there's no updated test-suite or anything, but I figured I'd dump it here anyhow.
